Pages/EditCharityEvent.razor

```diff
@@ -126,34 +108,74 @@
     KarmaContext KarmaContext = new KarmaContext();
     public CharityEvent CharityEvent;
     private string FilterValue { get; set; } = "";
-    private String nameSurname { get; set; } = "";
-    private String surnameName { get; set; } = "";
+    private string nameSurname { get; set; } = "";
+    private string surnameName { get; set; } = "";
+    private string ErrorMessage { get; set; } = "";
+    private int VolunteerCount { get; set; }
     bool panelOpenState1 = true;
     bool panelOpenState2 = true;
 
     string CurrentUserId { get; set; }
 
+    private void OnFocusHandler(FocusEventArgs e)
+    {
+        ErrorMessage = "";
+    }
+
+    private void OnFocusOutHandler(FocusEventArgs e)
+    {
+        var value = VolunteerCount;
+        if (value > 0 && value >= CharityEvent.Volunteers.Count)
+        {
+            CharityEvent.MaxVolunteers = value;
+        }
+        else if (value < 0)
+        {
+            ErrorMessage = "Illegal volunteer count, please choose a positive number.";
+        }
+        else
+        {
+            ErrorMessage = "Illegal volunteer count, please choose a value grater than currently assigned volunteeer count or reduce the number of volunteers in this event.";
```

Comment:
imho this looks cleaner
        if (value > 0)
        {
			if (value >= CharityEvent.Volunteers.Count)
				ErrorMessage = "Illegal volunteer count, please choose a value greater than currently assigned volunteeer count or reduce the number of volunteers in this event.";
            else
				CharityEvent.MaxVolunteers = value;
        }
        else
            ErrorMessage = "Illegal volunteer count, please choose a positive number.";
            
            also typo in word greater

Reply:
yeah it looks better this way, changed, fixed typo